### PR TITLE
Update linters configuration

### LINKED ISSUE #23: Replace `pydocstyle` with `flake8-docstrings` in the `pre-commit` check
Replace `pydocstyle` with `flake8-docstrings` in the `pre-commit` check

### PR DESCRIPTION
**Pull request by OlenaYefymenko**
- Edit pre-commit configuration for correct operation of the linters;
- Integrate yamlint into pre-commit configuration to ensure the YAML files adhere to specified guidelines;
- Fix warning raised by yamllint in the CI workflow configuration

Resolves #23

### CODE REVIEW

**Comment by webknjaz** [comment]
> Edit pre-commit configuration for correct operation of the linters;

It's rather obvious that commits would contain file edits to achieve something. But it'd be better to write down what the motivation and the effect of the changes is. Saying "Do something so it's done" doesn't provide useful or accurate details. Add some.

**Comment by webknjaz** [comment]
> Fix warning raised by yamllint in the CI workflow configuration

What warning? What's fixed? How? Why?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,17 +10,15 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Fetch the src
```
Comment (webknjaz) [inline .github/workflows/ci.yml:12]: Why did you add indentation instead of removing it? Why didn't you include a yamllint config as a part of the integration?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,35 +16,50 @@ repos:
       - id: name-tests-test
       - id: check-ast
       - id: debug-statements
+
```
Comment (webknjaz) [inline .pre-commit-config.yaml:19]: This doesn't look like something yamllint would require, it doesn't seem related to the change.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,35 +16,50 @@ repos:
...
     rev: v1.4.2
     hooks:
       - id: remove-tabs
+
   - repo: https://github.com/pycqa/pydocstyle
     rev: 6.3.0
     hooks:
       - id: pydocstyle
         exclude: ^tests/.*
-  - repo: https://github.com/PyCQA/pylint
```
Comment (webknjaz) [inline .pre-commit-config.yaml:35]: Reconfiguring unrelated linters isn't related to integrating yamllint. Formatting changes must always be separate from the functional ones.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -15,35 +16,50 @@ repos:
...
-          - --disable=C0330,R0913,R0914
+        name: pylint
+        entry: pylint
+        language: system
+        types: [python]
+        require_serial: true
+
   - repo: https://github.com/PyCQA/flake8
-    rev: 3.9.2
+    rev: 6.0.0
```
Comment (webknjaz) [inline .pre-commit-config.yaml:51]: Changing linter versions is a functional change. Integrating yamllint should be about formatting.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -15,35 +16,50 @@ repos:
...
     hooks:
       - id: flake8
+        additional_dependencies:
+          - flake8-docstrings
```
Comment (webknjaz) [inline .pre-commit-config.yaml:55]: This linter is already integrated as another hook. Of you want to replace it, drop the duplicate entry. Of course, such changes don't belong within formatting patches.

**Review by webknjaz — CHANGES_REQUESTED** [review]

**Comment by webknjaz** [comment]
> Update linters configuration

This title is very broad and doesn't accurately refer to anything useful. Rephrase it to say one exact thing the PR is supposed to achieve.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (OlenaYefymenko) [reply .github/workflows/ci.yml:12]: Ok, I will fix it. So, should I add  `yamllint` to `ci.yml` too?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (OlenaYefymenko) [reply .pre-commit-config.yaml:19]: Could you, please, clarify? Do you mean empty row?

File: .pre-commit-config.yaml (hunk 3 above)
Comment (OlenaYefymenko) [reply .pre-commit-config.yaml:35]: As far as I understand, I need to separate one PR for integrate `yamlint` and formatting changes can refer to this PR. The second PR should refer to fix `pylint` and `flake8` in `pre-commit tool`?

Comment (OlenaYefymenko) [reply .pre-commit-config.yaml:35]: Also, I have question about `pylint` in CI job. Firstly, I tried to use `.pylintrc`, but have errors with this config, then I looked another implementation in repo of Marta. But, in your repo `cheroot`  I see this config (`.pylintrc`). So, what the significant difference in these 2 implementations?

**Comment by OlenaYefymenko** [comment]
> > Update linters configuration
> 
> This title is very broad and doesn't accurately refer to anything useful. Rephrase it to say one exact thing the PR is supposed to achieve.

Ok, I will try. But should I avoid specifying the exact filename? Did I understand correctly?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (webknjaz) [reply .pre-commit-config.yaml:19]: Tables have rows. This is a line. But yes, I mean that adding an empty line doesn't seem to be related to changing yamllint config. It seems to be a formatting change while the PR states that it's targeted at changing some unspecified linter configuration.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (webknjaz) [reply .pre-commit-config.yaml:35]: > As far as I understand, I need to separate one PR for integration `yamlint` and formatting changes can refer to this PR.

In if the formatting changes are directly related to the yamllint configuration, they should be bundled. If they aren't related to linters but are your personal preference, it's probably keep them separate.

Comment (webknjaz) [reply .pre-commit-config.yaml:35]: > The second PR should refer to fix `pylint` and `flake8` in `pre-commit tool`?

Only if these are broken in the same way and have the same root of issue. If they aren't related, there's no reason why one would put them into the same patch. Instead, it'd best to make the changes atomic.

**Comment by webknjaz** [comment]
> But should I avoid specifying the exact filename?

It's usually unnecessary, unless how the file is named is an important detail. This probably depends on what you want the PR to be about. What's one end-goal? Is it about renaming a file? Is it about configuring checks/linters in some way? Is it important enough to be in the title as opposed to the long description?

> Did I understand correctly (I remember this recommendation from previous PR?)

This depends on whether it's an important detail that would improve accuracy of the title while not referring to unimportant low-level implementation details. The title is supposed to be high-level and accurate. If I get email notification about 5 PRs saying "update linters configuration", I'm not going to understand what's that all about. It's not compelling to open each of such notifications and wrestle the mind with what's the difference between those might be.

**Comment by webknjaz** [comment]
Why did you close this?

**Comment by webknjaz** [comment]
You could probably repurpose the PR to make changes related to pylint, or docstring linting, or version updates.

**Comment by OlenaYefymenko** [comment]
> Why did you close this?

Because, name of branch already not suit to current issue (`pylint`), I deleted old branch. Also, a lot of not relevant commits.

**Comment by webknjaz** [comment]
Well, most people don't look into the branch name anyway. So it's usually acceptable to iterate on things. The context is in the comments, by losing the context, you now have to keep track of two locations to react to things asked elsewhere in your new PR. I'd say this introduces unnecessary cognitive load, a mental overhead that could've been avoided.

**Comment by webknjaz** [comment]
By the way, it's possible to reopen PRs after having resurrected the branch to its original state (name + the same commit at head).

**Comment by OlenaYefymenko** [comment]
> By the way, it's possible to reopen PRs after having resurrected the branch to its original state (name + the same commit at head).

Ok.  I'll do it. 
Also, I noticed in your repo some useful settings for `flake8`
https://github.com/cherrypy/cheroot/blob/00c3cc3e5331fbaa10c44e5080899f1f3d9da6e7/.flake8
Can I focus on this example for my project? I mean what understandable to me

**Comment by OlenaYefymenko** [comment]
> By the way, it's possible to reopen PRs after having resurrected the branch to its original state (name + the same commit at head).

I tried, but couldn't reopen it. Additionally, I need the latest changes from the `main` branch, which pertain to edits in YAML files.

**Comment by webknjaz** [comment]
You can reopen it only if you recover the exact branch state with the same head commit.
